### ISSUE TITLE
Assume role prometheus assume role responses metric (per role)

### ISSUE THREAD

**Issue by max-lobur** [uswitch/kiam#189]
Like `kiam_metadata_responses_total`, but with assumed role name. Maybe update existing metrics to show role name.

With this, I will be able to find out top kiam traffic (assume-role) consumers and find non-optimized apps in my k8s cluster.

**Comment by pingles** [comment]
My concern would be that with things like CloudFormation you can end up with role names that feature some kind of unique suffix; including these role names as labels could potentially result in an extremely [high cardinality time series](https://prometheus.io/docs/practices/naming/).

You could achieve what you want currently through increasing the log level- Kiam servers will record JSON data that includes any annotated roles and the agents will record what's being accessed; assuming you're using something like ElasticSearch it should be relatively easy to use that to drive to the same result?

**Comment by max-lobur** [comment]
Cool, indeed. Yes I have ELK stack, I will try.

I was basing my request on metrics like `container_cpu_load_average_10s` which have pod name inside. This produces lots pf metrics I agree, but still works very fast and shows up on a CPU dashboard for the whole cluster:
![image](https://user-images.githubusercontent.com/5246026/49213656-70bbc800-f3cd-11e8-8af2-e9c7be20d82d.png)

**Comment by pingles** [comment]
Yep, that's fair. Ok, I'd happily take a PR that adds support (although it'd probably mean a new metric- not all `kiam_metadata_responses_total` would involve a role). So it'd probably need a new `kiam_metadata_credentials_responses_total` counter with the role name label.
